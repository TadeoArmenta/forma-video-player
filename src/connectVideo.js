import {connect} from 'react-redux'
import {bindActionCreators} from 'redux'

// A simplified syntax for connecting to the redux state for the video player
export const connectVideo = (
  videoState = [],
  videoActions = {},
  mergeProps,
  options
) => {

  return (WrappedComponent) => {
    return connect(
      (state) => ({player: videoState.reduce((result, key) => ({...result, [key]: state[key]}), {})}),
      (dispatch) => ({actions: bindActionCreators(videoActions, dispatch)}),
      mergeProps,
      options
    )(WrappedComponent)
  }
}
