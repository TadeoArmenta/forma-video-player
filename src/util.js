import React from 'react'
import createReactClass from 'create-react-class'
import {merge, pick} from 'lodash'
import {State} from 'replate'

const ReactComponentPrototype = React.Component.prototype
const ReactClassComponentPrototype = (Object.getPrototypeOf(Object.getPrototypeOf(new (createReactClass({ render () {} }))())))

export const isReactComponent = (Class) => {
  return Class ? (ReactComponentPrototype.isPrototypeOf(Class.prototype) || ReactClassComponentPrototype.isPrototypeOf(Class.prototype)) : false
}

export const makeStyles = (defaultStyles = {}, styles = {}) => {
  return merge(defaultStyles, pick(styles, Object.keys(defaultStyles)))
}

export const makeState = (name, values) => {
  return new State(name, {}, values.reduce((result, valueConfig) => ({
      ...result,
      [valueConfig.name]: new State(valueConfig.name, valueConfig.initial, {
        [valueConfig.name]: (state, action) => action.payload
      })
    }), {}
  ))
}
