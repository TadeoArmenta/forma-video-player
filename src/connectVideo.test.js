import React, {Component} from 'react'
import configureStore from 'redux-mock-store'
import ShallowTestRenderer from 'react-test-renderer/shallow'
import TestRenderer from 'react-test-renderer'
import {Provider} from 'react-redux'

import {connectVideo} from './connectVideo'
import * as util from './util'

describe('connectVideo', () => {

  class TestComponent extends Component {render() {return null}}
  const mockState = {player:{test: true, styles:{}}}
  const mockStore = configureStore()
  let store, renderer

  beforeEach(() => {
    store = mockStore(mockState)
    renderer = new ShallowTestRenderer()
  })

  it('should have redux props', () => {
    const Comp = connectVideo(['test'], {})(TestComponent)

    const rendered = renderer.render(
      <Comp store={store}/>
    )

    expect(rendered.props.player).toMatchSnapshot()
  })

  it('should connect with redux dispatch', () => {
    const actionBody = {type:'test', payload:'test-payload'}
    const mockActionCreator = () => actionBody

    const Comp = connectVideo([], {mockActionCreator})(TestComponent)

    const rendered = renderer.render(
      <Comp store={store}/>
    )

    const actions = rendered.props.actions

    expect(actions).toMatchSnapshot()
    actions.mockActionCreator()
    expect(store.getActions()).toEqual([actionBody])
  })
})
