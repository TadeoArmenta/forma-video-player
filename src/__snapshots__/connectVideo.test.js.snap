// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`connectVideo should connect with redux dispatch 1`] = `
Object {
  "mockActionCreator": [Function],
}
`;

exports[`connectVideo should have redux props 1`] = `
Object {
  "test": undefined,
}
`;
